### PR TITLE
adm: Add command `morph netmap-candidates`

### PR DESCRIPTION
**Pull request by acid-ant**
Close #1889

Signed-off-by: Anton Nikiforov <an.nikiforov@yadro.com>

### CODE REVIEW

File: cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+)
+
+var netmapCandidatesCmd = &cobra.Command{
+	Use:   "netmap-candidates",
+	Short: "List netmap candidates nodes",
+	PreRun: func(cmd *cobra.Command, _ []string) {
+		_ = viper.BindPFlag(endpointFlag, cmd.Flags().Lookup(endpointFlag))
+		_ = viper.BindPFlag(alphabetWalletsFlag, cmd.Flags().Lookup(alphabetWalletsFlag))
+	},
+	RunE: listNetmapCandidatesNodes,
```
Comment (carpawell) [inline cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go:21]: lets try not to use `RunE` according to #623?

File: cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+var short bool
+
+func init() {
+	netmapCandidatesCmd.Flags().StringP(endpointFlag, "r", "", "N3 RPC node endpoint")
+	netmapCandidatesCmd.Flags().String(alphabetWalletsFlag, "", "Path to alphabet wallets dir")
```
Comment (carpawell) [inline cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go:28]: why do we need it?

File: cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	netmapCandidatesCmd.Flags().BoolVar(&short, "short", false, "Shortens output of node info")
+}
+
+func listNetmapCandidatesNodes(cmd *cobra.Command, _ []string) error {
+	wCtx, err := newInitializeContext(cmd, viper.GetViper())
```
Comment (carpawell) [inline cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go:33]: that context is usually used for complex transactions. for RO purposes it looks too heavy, see `morph dump-config` as an example

File: cmd/frostfs-adm/internal/modules/morph/root.go [hunk 4]
```diff
@@ -323,4 +323,6 @@ func init() {
 	depositNotaryCmd.Flags().String(walletAccountFlag, "", "Wallet account address")
 	depositNotaryCmd.Flags().String(refillGasAmountFlag, "", "Amount of GAS to deposit")
 	depositNotaryCmd.Flags().String(notaryDepositTillFlag, "", "Notary deposit duration in blocks")
+
+	RootCmd.AddCommand(netmapCandidatesCmd)
```
Comment (carpawell) [inline cmd/frostfs-adm/internal/modules/morph/root.go:327]: why have you decided to init flags in a separate `init` and not like it is already done here?

File: pkg/core/netmap/prettyprint.go [hunk 5]
```diff
@@ -1,4 +1,4 @@
-package common
+package netmap
```
Comment (carpawell) [inline pkg/core/netmap/prettyprint.go:1]: could you, please, make that move in a separate commit?

Comment (acid-ant) [reply pkg/core/netmap/prettyprint.go:1]: In this case I need to add in `adm` dependency on `cli` or duplicate code. Is It ok?

File: cmd/frostfs-adm/internal/modules/morph/root.go (hunk 4 above)
Comment (acid-ant) [reply cmd/frostfs-adm/internal/modules/morph/root.go:327]: The idea was to reduce amount of lines here, also we  are using the same approach in `cli`.

File: cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go (hunk 1 above)
Comment (acid-ant) [reply cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go:21]: Refactored.

File: cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go (hunk 2 above)
Comment (acid-ant) [reply cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go:28]: Removed.

File: cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go (hunk 3 above)
Comment (acid-ant) [reply cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go:33]: Refactored.

File: pkg/core/netmap/prettyprint.go (hunk 5 above)
Comment (acid-ant) [reply pkg/core/netmap/prettyprint.go:1]: Updated.

File: cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+	cs, err := c.GetContractStateByID(1)
+	utilCore.ExitOnErr(cmd, "can't get NNS contract info: %w", err)
+
+	nmHash, err := nnsResolveHash(inv, cs.Hash, netmapContract+".frostfs")
+	utilCore.ExitOnErr(cmd, "can't get netmap contract hash: %w", err)
+
+	res, err := inv.Call(nmHash, "netmapCandidates")
+	utilCore.ExitOnErr(cmd, "can't fetch list of network config keys from the netmap contract", err)
+	if res != nil {
```
Comment (fyrchik) [inline cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go:26]: Is it possible to have `res == nil` and `err == nil` at the same time?

File: cmd/frostfs-adm/internal/modules/morph/root.go [hunk 7]
```diff
@@ -323,4 +335,8 @@ func init() {
...
+	netmapCandidatesCmd.Flags().StringP(endpointFlag, "r", "", "N3 RPC node endpoint")
+	netmapCandidatesCmd.Flags().BoolVar(&short, "short", false, "Shortens output of node info")
```
Comment (fyrchik) [inline cmd/frostfs-adm/internal/modules/morph/root.go:341]: Why not short by default and `--verbose` for everything?

File: cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go (hunk 6 above)
Comment (acid-ant) [reply cmd/frostfs-adm/internal/modules/morph/netmap_candidates.go]: Removed redundant check.

File: cmd/frostfs-adm/internal/modules/morph/root.go (hunk 7 above)
Comment (acid-ant) [reply cmd/frostfs-adm/internal/modules/morph/root.go]: Good idea, replaced with `--verbose`.

File: pkg/core/netmap/prettyprint.go (hunk 5 above)
Comment (acid-ant) [reply pkg/core/netmap/prettyprint.go]: Moved from `pkg` to `cmd/internal/common`

File: cmd/frostfs-cli/internal/client/sdk.go [hunk 8]
```diff
@@ -10,6 +10,7 @@ import (
 
 	"github.com/TrueCloudLab/frostfs-node/cmd/frostfs-cli/internal/common"
 	"github.com/TrueCloudLab/frostfs-node/cmd/frostfs-cli/internal/commonflags"
+	commonCmd "github.com/TrueCloudLab/frostfs-node/cmd/internal/common"
```
Comment (fyrchik) [inline cmd/frostfs-cli/internal/client/sdk.go:13]: There is a typo in the commit message.

File: cmd/frostfs-adm/internal/modules/root.go [hunk 9]
```diff
@@ -34,7 +33,9 @@ func init() {
 	// use stdout as default output for cmd.Print()
 	rootCmd.SetOut(os.Stdout)
 
-	rootCmd.PersistentFlags().StringP(configFlag, "c", "", "Config file")
+	rootCmd.PersistentFlags().StringP(commonflags.ConfigFlag, commonflags.ConfigFlagShorthand, "", commonflags.ConfigFlagUsage)
+	rootCmd.PersistentFlags().BoolP(commonflags.Verbose, commonflags.VerboseShorthand, false, commonflags.VerboseUsage)
+	_ = viper.BindPFlag(commonflags.Verbose, rootCmd.PersistentFlags().Lookup(commonflags.Verbose))
```
Comment (fyrchik) [inline cmd/frostfs-adm/internal/modules/root.go:38]: Can we factor out common flags in a separate commit? And add `netmap-candidates` in the next one.

Comment (acid-ant) [reply cmd/frostfs-adm/internal/modules/root.go:38]: Done.

File: cmd/frostfs-cli/internal/client/sdk.go (hunk 8 above)
Comment (acid-ant) [reply cmd/frostfs-cli/internal/client/sdk.go:13]: Thanks, fixed.

**Review by carpawell — APPROVED** [review]

**Review by fyrchik — APPROVED** [review]
